### PR TITLE
Better sync UI

### PR DESCRIPTION
**Pull request by mahaloz**
There are a lot of changes here:
- ContextMenu for functions to enable syncing
- Fix variable renaming not transferring across function switching
- Add variable renaming syncable
- Removed old sync ui

### CODE REVIEW

**Comment by mahaloz** [comment]
This PR corresponds to angr: https://github.com/angr/angr/pull/2549 and to the same branch name in internal projects.

File: angrmanagement/ui/main_window.py
Comment (ltfish) [inline angrmanagement/ui/main_window.py:225]: Commit this line as a separate PR (check if the current OS is macOS, and if so, call `setNativeMenuBar(False)`).

File: angrmanagement/ui/views/code_view.py
Comment (ltfish) [inline angrmanagement/ui/views/code_view.py:64]: This feature will conflict with my on-going development of decompilation manager. Wait until it is merged to master.

**Review by ltfish — CHANGES_REQUESTED** [review]

File: angrmanagement/ui/main_window.py
Comment (mahaloz) [reply angrmanagement/ui/main_window.py:225]: Got it. This was actually accidentally left here, but I will still make a change for it.

File: angrmanagement/ui/views/code_view.py
Comment (mahaloz) [reply angrmanagement/ui/views/code_view.py:62]: Will need to wait on this pr since this is not possible to merge now.

File: angrmanagement/ui/views/code_view.py
Comment (mahaloz) [reply angrmanagement/ui/views/code_view.py:64]: @ltfish then this whole PR will need to wait since variable renaming, as well as syncing, is broken without some sort of decompilation caching

**Review by mahaloz — COMMENTED** [review]
Not mergeable now.

**Comment by mahaloz** [comment]
This PR is now redundant since all the features aside from the variable renaming fix are in: https://github.com/angr/angr-management/pull/233. This can be closed, and the rename fix can be implemented in @ltfish's decompilation API.

**Pull request closed without merging**
